client/index.html

```diff
@@ -58,6 +58,7 @@ <h1 class="title">The Lounge is loading…</h1>
 				<div id="chat-container" class="window">
 					<div id="chat"></div>
 					<form id="form" method="post" action="">
+						<div id="connection-error"><span>Connection failed: <span id="connection-error-reason">??</span>. Refresh the page to reconnect.</span></div>
```

Comment:
Outer `<span>` is an accident here?

I'd definitely replace second sentence with `Click here to reconnect` and make it a link or a button (maybe even prepend this with http://fontawesome.io/icon/refresh/ ?).
Reason why we have to make it clickable is that, before your PR, app would refresh by itself when connection is back, or something like that. With your PR, app in standalone mode on mobile cannot be refreshed with pull-to-refresh (because of #618). So that means once connection is lost, user now has to fully close the app and open it again. Makes sense?

Also, I don't think there is any need for specific message. `Connection lost. Click here to reconnect` is enough, and most widespread messaging clients will display the same. Plus if for some reason Socket.io spits some weird looking message, that would make it even more confusing.
I just tried disabling wifi on my phone, and I got `Connection failed: transport close`. As a user, that doesn't tell me anything, but `Connection lost` seems more appropriate and generic enough for all cases.
Plus, you are logging to console anyway, if intent is to debug.